### PR TITLE
Build: fix build with clang compiler

### PR DESCRIPTION
**Pull request by fabbione**
Signed-off-by: Fabio M. Di Nitto <fdinitto@redhat.com>

### CODE REVIEW

**Comment by fabbione** [comment]
FYI clang build has been failing for weeks in CI. Should we move it as voting?

File: daemons/controld/controld_execd.c [hunk 1]
```diff
@@ -2666,6 +2666,7 @@ log_executor_event(const lrmd_event_data_t *op, const char *op_key,
         case PCMK_EXEC_CANCELLED:
             log_level = LOG_INFO;
             // Fall through
+	    __attribute__((fallthrough));
```
Comment (nrwahl2) [inline daemons/controld/controld_execd.c:2669]: Huh, I thought we found out there is no macro or compiler directive that works for both GCC and Clang. See also https://projects.clusterlabs.org/T558

I hope this works :)

File: daemons/controld/controld_join_dc.c [hunk 2]
```diff
@@ -526,7 +526,7 @@ finalize_sync_callback(xmlNode * msg, int call_id, int rc, xmlNode * output, voi
             int count_finalizable = 0;
 
             count_finalizable = crmd_join_phase_count(crm_join_integrated)
-                                + crmd_join_phase_count(crm_join_nack);
+                                + crmd_join_phase_count(crm_join_nack)
```
Comment (nrwahl2) [inline daemons/controld/controld_join_dc.c:529]: D'oh. Thanks!

File: daemons/controld/controld_execd.c (hunk 1 above)
Comment (fabbione) [reply daemons/controld/controld_execd.c:2670]: it does actually work with recent gcc, but it fails with gcc on rhel7. There are other workarounds for that.

File: daemons/controld/controld_join_dc.c (hunk 2 above)
Comment (fabbione) [reply daemons/controld/controld_join_dc.c:529]: no problem

**Comment by fabbione** [comment]
retest this please
(CI timeout for something completely unrelated to pcmk)

**Comment by fabbione** [comment]
manual builds:
https://ci.kronosnet.org/view/pacemaker/job/pacemaker-build-all-nonvoting-clang/3015/ (against corosync camelback)
https://ci.kronosnet.org/view/pacemaker/job/pacemaker-build-all-nonvoting-clang/3016/ (against corosync main)

I am not going to PR against 2.1. There are code changes that don´t cherry-pick and to be honest, clang has been failing for over 10 months. You guys can sort out how you want to handle it.

File: lib/common/output.c [hunk 3]
```diff
@@ -120,7 +120,7 @@ pcmk__register_formats(GOptionGroup *group,
 }
 
 void
-pcmk__unregister_formats() {
+pcmk__unregister_formats(void) {
```
Comment (kgaillot) [inline lib/common/output.c:123]: Don't know how this was missed in c0e04a6

**Review by kgaillot — APPROVED** [review]

**Pull request merged**